src/dopri.ts

```diff
@@ -155,13 +158,30 @@ export class Dopri implements Integrator {
         };
     }
 
+    public steps(): number[] {
+        return this._history.map((el) => el.t);
+    }
+
     private _reset() {
         this._nSteps = 0;
         this._nStepsAccepted = 0;
         this._nStepsRejected = 0;
         this._stiffNStiff = 0;
         this._stiffNNonstiff = 0;
         this._lastError = 0;
+        this._tcrit = [...this._control.tcrit];
+    }
+
+    private _nextTcrit(): number {
+        if (this._tcrit.length === 0) {
+            return Infinity;
+        }
+        const nextTcrit = this._tcrit[0];
+        if (nextTcrit > this._t) {
+            return nextTcrit;
+        }
+        this._tcrit = this._tcrit.slice(1);
+        return this._nextTcrit();
```

Comment:
Up to preference but this reduces the need for recursive calls and i guess a bit more verbose logic
```suggestion
        const this._tcrit = this._tcrit.filter(time => time > this._t);
        return this._tcrit[0] || Infinity;
```

Reply:
I think I'd prefer the first early exit clause to remain, but otherwise happy with a filter approach to remove recursion (though who does not like a bit of recursion here or there...)

Reply:
```
this._tcrit[0] || Infinity
```

this would also prevent a tcrit of 0 being allowed as it's falsy